src/server/services/home-block.service.ts

```diff
@@ -11,3 +24,132 @@ export const getHomeBlocks = <TSelect extends Prisma.HomeBlockSelect = Prisma.Ho
     orderBy: { index: 'asc' },
   });
 };
+
+export const getHomeBlockById = async ({
+  id,
+  ctx,
+  ...input
+}: GetHomeBlockByIdInputSchema & { ctx: Context }) => {
+  const homeBlock = await dbRead.homeBlock.findUniqueOrThrow({
```

Comment:
I've don't usually use (or never used 😛) `findUniqueOrThrow` anywhere, but I see the benefit here 👍